staging/src/k8s.io/apimachinery/pkg/api/meta/help.go

```diff
@@ -130,34 +144,35 @@ func EachListItem(obj runtime.Object, fn func(runtime.Object) error) error {
 	if len == 0 {
 		return nil
 	}
-	takeAddr := false
-	if elemType := items.Type().Elem(); elemType.Kind() != reflect.Pointer && elemType.Kind() != reflect.Interface {
-		if !items.Index(0).CanAddr() {
-			return fmt.Errorf("unable to take address of items in %T for EachListItem", obj)
-		}
-		takeAddr = true
-	}
 
 	for i := 0; i < len; i++ {
 		raw := items.Index(i)
-		if takeAddr {
-			raw = raw.Addr()
-		}
-		switch item := raw.Interface().(type) {
-		case *runtime.RawExtension:
-			if err := fn(item.Object); err != nil {
+		switch {
+		case raw.Type() == rawExtensionObjectPointerType:
+			if err := fn(raw.Interface().(*runtime.RawExtension).Object); err != nil {
 				return err
 			}
-		case runtime.Object:
-			if err := fn(item); err != nil {
+		case raw.Type() == objectType || raw.Type().Implements(objectType):
+			if err := fn(raw.Interface().(runtime.Object)); err != nil {
 				return err
 			}
 		default:
-			obj, ok := item.(runtime.Object)
-			if !ok {
-				return fmt.Errorf("%v: item[%v]: Expected object, got %#v(%s)", obj, i, raw.Interface(), raw.Kind())
+			if !alloc {
```

Comment:
invert this and make this block handle the new `alloc` case, leaving the existing code as-is

Reply:
actually, you separate `alloc` to its own `case` rather than the `default` needing to contain a nested `if` block

Reply:
resolved,  PTAL~